src/NuGet.Clients/NuGet.CommandLine/NuGetCommand.resx

```diff
@@ -5336,4 +5336,7 @@ nuget locals global-packages -list</value>
   <data name="RestoreCommandRecursive" xml:space="preserve">
     <value>Restore all referenced projects for UWP and NETCore projects. This does not include packages.config projects.</value>
   </data>
+  <data name="ForceRestoreCommand" xml:space="preserve">
+    <value>NuGet restore does not perform a full restore when the inputs have not changed. This option skips the that optimization.</value>
```

Comment:
This should mention that it is only for PackageReference/project.json and give some hints as to why this should be used. At the same time it needs to be clear that this isn't the same as -NoCache which is for the http cache only.

How about:
Forces all dependencies to be resolved even if the last restore was successful. This is equivalent to deleting project.assets.json. (Does not apply to packages.config)